### ISSUE TITLE
`__call` return type annotation is ignored on method calls

### ISSUE THREAD

**Issue by weirdan** [vimeo/psalm#895]
https://getpsalm.org/r/f3122cd4ba
```php
<?php
  class B {
     public function __call(string $methodName, array $args): string {
       return __METHOD__;
     }
  }
  class A {
     /** @return B */
     public function __call(string $methodName, array $args) {
       return new B;
     }
  }
  strlen((new A)->zugzug()); // expected: argument of strlen cannot be B, actual: ... cannot be mixed
  strlen((new A)->bar()->baz()); // expected: no errors, actual: ... cannot be mixed
```
39870f97c46b85e88f3ea82703bbe298ba904aa7 has not fixed this.

Real-world example: https://github.com/phpspec/prophecy/blob/master/src/Prophecy/Prophecy/ObjectProphecy.php#L238

I understand this may conflict with `@method` annotations, but it would be useful to use `@return` at least when there's no `@method`s present.
